AIAgent/tests/test_statistics.py

```diff
@@ -0,0 +1,96 @@
+import csv
+import pytest
+from ml.training.statistics import get_svms_statistics
+from ml.training.dataset import Result
+from pathlib import Path
+from tests.utils import read_configs
+import yaml
+from common.classes import GameFailed, Map2Result, GameResult
+from run_training import get_maps
+from random import choice
+from common.config import ValidationConfig
+from unittest import mock
+import shutil
+import paths
+from paths import CURRENT_TABLE_PATH
+
+
+class TrainingDatasetMock:
+    def __init__(self, mock_maps_results: dict[str, Result]) -> None:
+        self.maps_results = mock_maps_results
+
+
+class TestSVMsStatistics:
+    def create_statistics_file_with_header(self, maps):
+        self.tmp_dir.mkdir(exist_ok=True)
+        with open(self.test_csv_file_path, "w") as statistics_file:
+            statistics_writer = csv.DictWriter(
+                statistics_file,
+                sorted([game_map2svm.GameMap.MapName for game_map2svm in maps]),
+            )
+            statistics_writer.writeheader()
+
+    def remove_tmp(self):
+        shutil.rmtree("tests/tmp")
+
+    @pytest.fixture(autouse=True)
+    def mock_variables_and_create_tmp(self, monkeypatch):
+        self.tmp_dir = Path("./tests/tmp")
+        self.test_csv_file_path = Path(self.tmp_dir / "test_statistics.csv")
+        monkeypatch.setattr(paths, "CURRENT_TABLE_PATH", self.test_csv_file_path)
+        yield
+        self.remove_tmp()
+
+    @pytest.fixture(params=read_configs("tests/resources/svms_validation_configs"))
+    def get_args(self, request):
+        with open(request.param) as file:
+            val_config = ValidationConfig(**yaml.safe_load(file)).validation
+
+        dataset = TrainingDatasetMock(
+            mock_maps_results={
+                "Map1": Result(100, -4, -9028, 5),
+                "Map2": Result(0, -1, -8, 0),
+            }
+        )
+        maps = get_maps(val_config)
+        return val_config, dataset, maps
+
+    def test_single_epoch_with_successful_results(self, get_args):
+        metrics = {
+            "average_coverage": 100.0,
```

Comment:
Can we generate names using svms?
Can we move svm-independent names to constants to reuse  [here](https://github.com/PySymGym/PySymGym/pull/120/files#diff-77b01a395331db59634890091292862d8a6cdbaaf4de890c56bbc4c67d845667R49)